seip00-sp2

voiran opened the pull request:
Link: https://eso.vse.cz/~seip00/sp2/
Zatím work in progress, tohle je základ

voiran committed:
seip00-sp2

Link: https://eso.vse.cz/~seip00/sp2/
Zatím work in progress, tohle je základ

voiran committed:
Verze 1.0

Přidání google Places API pro autocomplete,
Přidání možnosti uložení a odstranění města do local storage
celková úprava, změnilo se hodně

nvbach91 commented on www/seip00/sp2/index.html:
```diff
@@ -0,0 +1,57 @@
+    </div>
+
+    <script>
+        function activatePlacesSearch() {
```
tohle presunte do `main.js`

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+const body = document.querySelector('.body');
+const searchbox = document.querySelector('.search');
+const button = document.querySelector('.srchbutton');
+button.addEventListener("click", setQuery);
```
apostrof (vsude misto `"`)

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+const api = {
```
veskery kod zabalit do closure, napr `window.onload = () => {}`

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+}
+//Funkce pro zobrazení fetchnutých dat
+function displayResults(weather) {
+    let city = document.querySelector(".searched-place .city");
```
tyto elementy vybirejte predem a ve funkci na ne pouze odkazete

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+    weather_icon.innerHTML = `<img src="icons/${weather.weather[0].icon}.png"/>`
+}
+//Funkce na převedení data do češtiny
+function dateCreate(d) {
```
pouzijte knihovnu `moment.js`

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+    feel.innerText = `Pocitová teplota: ${Math.round(weather.main.feels_like)}°C`
+
+    let weather_icon = document.querySelector(".current .icon");
+    weather_icon.innerHTML = `<img src="icons/${weather.weather[0].icon}.png"/>`
```
nevalidni html

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+function dateCreate(d) {
+    const months = ["Ledna", "Února", "Března", "Dubna", "Května", "Června", "Července", "Srpna", "Září", "Října", "Listopadu", "Prosince"];
+    const days = ["Ne.", "Po.", "Út.", "St.", "Čt.", "Pá.", "So."];
+    let day = days[d.getDay()];
```
const

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+//Funkce pro uložení města do local storage
+function setSaved() {
+    const toSave = searchbox.value;
+    var splitCity = toSave.split(",", 1);
```
const

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+    const toSave = searchbox.value;
+    var splitCity = toSave.split(",", 1);
+    if (toSave.length != 0) {
+        localStorage.setItem(splitCity, toSave);
```
do localStorage se uklada JSON pod vhodne nazvanym klicem, tady to bude treba `localStorage.setItem('cities', '["prague", "london"]');`

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+    var splitItem = item.split(",", 1);
+
+    saved.innerHTML += `<p>${item}</p>`;
+    buttons.innerHTML += `<button onclick="deleteCity('${splitItem}')">Smazat</button>
```
vytvarejte tento button pomoci `document.createElement`

nvbach91 commented on www/seip00/sp2/js/main.js:
```diff
@@ -0,0 +1,90 @@
+
+    saved.innerHTML += `<p>${item}</p>`;
+    buttons.innerHTML += `<button onclick="deleteCity('${splitItem}')">Smazat</button>
+    <button onclick="loadCity('${splitItem}')">Načíst</button>`;
```
a nepouzivejte zadny inline JS

voiran committed:
Verze 2.0

Upraveno vše dle domluvy na obhajobě, kromě google places Autocomplete. Nepřišel jsem na to jak ho dát mimo html, i všechny návody co jsem našel jsou takto napsané, jinak vše hotovo.
V CSS jsem ještě přidal "animaci" kliknutí na button a změnil kurzor myši při najetí na button.

nvbach91 reviewed (approved):
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)
